docs/ai/shade-agents/introduction.md

```diff
@@ -0,0 +1,94 @@
+
+Current Web3 agents fall into one of two camps:
+1. They are trustless and verifiable by creating and using a private key within a trusted execution environment (TEE), but if the TEE goes down, the accounts and funds are lost.
+2. The accounts are persistent, but the agents are centralized.
+
+Shade Agents provide verifiability, non-custody, and persistent accounts by operating in TEEs and implementing decentralized key management. Any agent running with the same code inside a TEE can access the same accounts.
+
+These agents can autonomously sign transactions across any chain, interact with AI models and external data sources, manage assets, and perform privacy-preserving, verifiable computations, offering the flexibility and performance of Web2 with the verifiability of Web3.
+
+:::info
+Shade Agents power [Agentic Protocols](./examples.md#agentic-protocols): a new type of decentralized application designed to be autonomous, proactive, and intelligent.
+:::
+
+---
+
+## How do Shade Agents work?
+
+Shade Agents consist of two main components: a `worker agent` and an `agent smart contract`.
+
+![Shade Agent Architecture](/docs/assets/shade-agents/shade-agent-stack-diagram.png)
+
+When a worker agent is booted up in a TEE, the `TEE's hardware-based entropy` generates a random private key and account. This private key is exclusively used to call the agent contract, not for storing funds.
+
+The worker agent calls the `register_worker` function on the agent contract, providing two pieces of information generated inside the TEE:
+- A `remote attestation quote` (which proves it is running inside a genuine TEE).
+- The Docker image's SHA256 `code hash` (to prove that the expected code is running).
+
+If the attestation quote is valid and the code hash matches the expected code hash of the worker agent (defined during agent contract deployment), the worker agent's accounts are added to a list of valid worker agents.
+
+Once registered, the worker agent can call the `sign_tx` function on the agent contract, enabling it to sign transactions on behalf of the Shade Agent. The Shade Agent utilizes [chain signatures](../../chain-abstraction/chain-signatures.md) for decentralized key management, allowing it to hold assets and sign transactions on nearly any chain.
+
+`Anyone` can deploy the same Docker image of the worker agent to a TEE and have it register as a valid worker agent and access the sign_tx function to use the same accounts. This means the accounts are never lost and the service is `always live`. Worker agents are designed to be stateless to facilitate this functionality.
+
+<details>
+
+<summary>What is a TEE?</summary>
```

Comment:
maybe this goes when we first say TEE, or closer to the first time we say TEE on this sub-section